app/scripts/modules/ecs/src/serviceDiscovery/serviceDiscovery.read.service.ts

```diff
@@ -0,0 +1,16 @@
+import { module } from 'angular';
+
+import { API } from '@spinnaker/core';
+import { IServiceDiscoveryRegistryDescriptor } from './IServiceDiscovery';
+
+export class ServiceDiscoveryReader {
+  public listServiceDiscoveryRegistries(): ng.IPromise<IServiceDiscoveryRegistryDescriptor[]> {
+    return API.all('ecs')
+      .all('serviceDiscoveryRegistries')
+      .getList();
+  }
+}
+
+export const ECS_SERVICE_DISCOVERY_READ_SERVICE = 'spinnaker.ecs.servicediscovery.read.service';
+
+module(ECS_SERVICE_DISCOVERY_READ_SERVICE, []).service('serviceDiscoveryReader', ServiceDiscoveryReader);
```

Comment:
minor thing - you don't really need this to be an Angular service...if you make the one method static, you can slim the whole thing down to:
```typescript
import { IPromise } from 'angular';
import { API } from '@spinnaker/core';
import { IServiceDiscoveryRegistryDescriptor } from './IServiceDiscovery';

export class ServiceDiscoveryReader {
  public static listServiceDiscoveryRegistries(): IPromise<IServiceDiscoveryRegistryDescriptor[]> {
    return API.all('ecs')
      .all('serviceDiscoveryRegistries')
      .getList();
  }
}
```
then, where you need it, just import the service and call the method a la `ServiceDiscoveryReader.listServiceDiscoveryRegistries()`.